### PR TITLE
[ncl] Add BrightnessScreen which uses expo-brightness

### PR DESCRIPTION
**Pull request by hesyifei**
# Why

Currently native-component-list does not have a screen to demonstrate the Brightness API ([expo-brightness](https://github.com/expo/expo/tree/master/packages/expo-brightness)).

# How

iOS demo
Android demo

# Test Plan
It works as expected on iOS and Android.

### CODE REVIEW

File: apps/native-component-list/src/screens/BrightnessScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  render() {
+    let views = brightnessTypes.map(type => {
+      return (
+        <View key={type} style={{ padding: 20 }}>
+          <HeadingText>{type}</HeadingText>
+          {type === 'SystemBrightness' && (
+            <Button
+              title="Permissions.SYSTEM_BRIGHTNESS"
+              onPress={() => {
+                Permissions.askAsync(Permissions.SYSTEM_BRIGHTNESS).then(result => {
```
Comment (tsapeta) [inline apps/native-component-list/src/screens/BrightnessScreen.tsx]: Can you convert all those functions inside `render` to be class members? That would be much more clear I guess 🤔

File: apps/native-component-list/src/screens/BrightnessScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+export default class BrightnessScreen extends React.Component<{}, State> {
+  static navigationOptions = {
+    title: 'Brightness',
+  };
+
+  readonly state: State = {
+    initBrightness: {},
+    sliderBrightness: {},
+    systemBrightnessPermissionGranted: true,
```
Comment (tsapeta) [inline apps/native-component-list/src/screens/BrightnessScreen.tsx]: Shouldn't it default to `false`? 😛

File: apps/native-component-list/src/screens/BrightnessScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      this.setState({ initBrightness: { ...this.state.initBrightness, Brightness: value } });
+    });
+
+    Brightness.getSystemBrightnessAsync().then(value => {
+      this.setState({ initBrightness: { ...this.state.initBrightness, SystemBrightness: value } });
+    });
+
+    Permissions.getAsync(Permissions.SYSTEM_BRIGHTNESS).then(result => {
+      if (result.status !== 'granted') {
+        this.setState({ systemBrightnessPermissionGranted: false });
```
Comment (tsapeta) [inline apps/native-component-list/src/screens/BrightnessScreen.tsx]: It might be better to make it `true` once `granted`

**Review by tsapeta — CHANGES_REQUESTED** [review]
Thanks! Looks good, just a few small comments below 👇

File: apps/native-component-list/src/screens/BrightnessScreen.tsx (hunk 1 above)
Comment (hesyifei) [reply apps/native-component-list/src/screens/BrightnessScreen.tsx]: Thanks for the suggestions! I've fixed them 😄

**Review by tsapeta — APPROVED** [review]
Looks good, thanks! 💪

**Pull request merged**